src/embed/sync.rs

```diff
@@ -0,0 +1,128 @@
+use super::http as embed_http;
```

Comment:
I thought you wanted to do `sync` as a peer to `db`, e.g., `src/sync/pull.rs`. I liked that plan.

Reply:
i perceived the embed::sync module as that kind of peer but happy to move it. should i think of embed as the embedded api, where the api implementations generally live elsewhere?

Reply:
> I guess there's really no difference it's just directory structure. But so
far we only have a single-level flat directory structure. Conceptually I
was thinking of everything inside of 'embed' having to do with embedding,
and since sync stands alone, I expected to see it outside of embed as its
own top level dir under src.

I'll move it in a future PR.